File: ts-sdk/src/sdk/models/operations/getademotionrequest.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Type } from "class-transformer";


export class GetADemotionRequestPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=database" })
  database: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=name" })
  name: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=organization" })
  organization: string;
}

export class GetADemotionRequestRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetADemotionRequestPathParams;
}

export class GetADemotionRequestResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getADemotionRequest200ApplicationJSONObject?: Record<string, Record<string, any>>;
}

File: go-sdk/pkg/models/operations/getademotionrequest.go
package operations

type GetADemotionRequestPathParams struct {
	Database     string `pathParam:"style=simple,explode=false,name=database"`
	Name         string `pathParam:"style=simple,explode=false,name=name"`
	Organization string `pathParam:"style=simple,explode=false,name=organization"`
}

type GetADemotionRequestRequest struct {
	PathParams GetADemotionRequestPathParams
}

type GetADemotionRequestResponse struct {
	ContentType                                 string
	StatusCode                                  int
	GetADemotionRequest200ApplicationJSONObject map[string]map[string]interface{}
}
